cmd/main.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	kitlog "github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"github.com/jarri-abidi/todolist/config"
+	"github.com/jarri-abidi/todolist/inmem"
+	"github.com/jarri-abidi/todolist/todolist"
+)
+
+func main() {
+	logger := kitlog.NewLogfmtLogger(os.Stderr)
+
+	conf, err := config.Load(".", "app.env")
+	if err != nil {
+		logger.Log("msg", "could not load config", "err", err)
+	}
+
+	var service todolist.Service
+	service = todolist.NewService(inmem.NewTodoStore())
+	service = todolist.NewLoggingService(logger, service)
+
+	mux := http.NewServeMux()
+	mux.Handle("/todolist/v1/", todolist.MakeHandler(service, logger))
+	mux.Handle("/metrics", promhttp.Handler())
+
+	server := &http.Server{
+		Addr:         conf.ServerAddress,
+		WriteTimeout: conf.ServerWriteTimeout,
+		ReadTimeout:  conf.ServerReadTimeout,
+		IdleTimeout:  conf.ServerIdleTimeout,
+		Handler:      mux,
+	}
+
+	fmt.Println(`
+	 _____          _           __ _     _   
+   	/__   \___   __| | ___     / /(_)___| |_ 
+	  / /\/ _ \ / _  |/ _ \   / / | / __| __|
+	 / / | (_) | (_| | (_) | / /__| \__ \ |_ 
+	 \/   \___/ \__,_|\___/  \____/_|___/\__|
+	`)
+
+	quit := make(chan os.Signal)
+	signal.Notify(quit, syscall.SIGINT)
+	go func() {
+		<-quit
+		if err := server.Shutdown(context.Background()); err != nil {
+			logger.Log("terminated", err)
+		}
+	}()
+
+	logger.Log("transport", "http", "address", conf.ServerAddress, "msg", "listening")
+	logger.Log("terminated", server.ListenAndServe())
```

Comment:
don't you think we should defer this line?

Reply:
defer to run the server? not sure how that would help. maybe you could share a code suggestion to be more precise.

Reply:
I'm sorry. I focused on `"terminated"` word and didn't check `server.ListenAndServe()` on line 61. But again, instead of logging on line 61, we could simply write `server.ListenAndServe()` because "terminated" on line 61 isn't getting logged.

Reply:
it is actually getting logged once the server shuts down:
<img width="708" alt="image" src="https://user-images.githubusercontent.com/8291818/189531668-0666fcb1-c37d-4082-9c5f-441e22c0ed06.png">

Reply:
Got it. Let's discuss it on our next call. I'm bit confused with `terminated` in goroutine and `terminated` on line 61.